cypress/utils/apps/query_enhancements/saved_queries.js

```diff
@@ -0,0 +1,602 @@
+      true
+    );
+  }
+
+  cy.setQueryEditor(queryString, { parseSpecialCharSequences: false });
+
+  if (histogram) {
+    cy.getElementByTestId('discoverIntervalSelect').select('w');
+  }
+};
+
+/**
+ * Verify that only DQL and Lucene saved queries are shown in the Visualize page.
+ * @param {boolean} savedQueriesNewUIEnabled - Has this flag been enabled (default is true)
+ */
+export const verifyValidSavedQueriesShownOnVisualize = (savedQueriesNewUIEnabled = true) => {
+  if (savedQueriesNewUIEnabled) {
+    // Currently functionality has not been implemented for savedQueriesNewUI
+  } else {
+    cy.getElementByTestId('createVisualizationPromptButton').click();
+    cy.getElementByTestId('visType-table').click();
+    cy.getElementByTestId('savedObjectFinderItemList')
+      .findElementByTestIdLike('savedObjectTitledata')
+      .click();
+
+    cy.getElementByTestId('showFilterActions').click();
+    cy.getElementByTestId('savedQueries').click();
+
+    cy.getElementByTestId('save-query-panel').contains('DQL');
+    cy.getElementByTestId('save-query-panel').contains('Lucene');
+    cy.getElementByTestId('save-query-panel').contains('OpenSearch SQL').should('not.exist');
+    cy.getElementByTestId('save-query-panel').contains('PPL').should('not.exist');
+  }
+};
+
+/**
+ * Verify query does not exist in saved queries.
+ * @param {string} deletedQueryName - Name of the query that should not exist.
+ * @param {boolean} savedQueriesNewUIEnabled - Has this flag been enabled (default is true)
+ */
+export const verifyQueryDoesNotExistInSavedQueries = (
+  deletedQueryName,
+  savedQueriesNewUIEnabled = true
+) => {
+  cy.reload();
+  if (savedQueriesNewUIEnabled) {
+    cy.getElementByTestId('saved-query-management-popover-button').click();
+    cy.getElementByTestId('saved-query-management-open-button').click();
+    cy.getElementByTestId('savedQueriesFlyoutBody').contains(deletedQueryName).should('not.exist');
+    // Two references to two buttons layered over each other.
+    cy.getElementByTestId('euiFlyoutCloseButton').first().click({ force: true });
+  } else {
+    cy.getElementByTestId('saved-query-management-popover-button').click();
+    cy.getElementByTestId('osdSavedQueryManagementList')
+      .contains(deletedQueryName)
+      .should('not.exist');
+    cy.getElementByTestId('saved-query-management-popover-button').click();
+  }
+};
+
+/**
+ * Update and save the saved query with alternate config, then verify has been updated correctly.
+ *  @param {SavedSearchTestConfig} config - the config for the test case to be updated
+ * @param {boolean} savedQueriesNewUIEnabled - Has this flag been enabled (default is true)
+ */
+export const updateAndVerifySavedQuery = (config, savedQueriesNewUIEnabled = true) => {
+  // Create alternate config
+  const alternateConfig = generateAlternateTestConfiguration(config);
+  cy.loadSaveQuery(config.saveName, savedQueriesNewUIEnabled);
+
+  // wait for saved query to load
+  cy.wait(2000);
```

Comment:
Is there an element we can wait for instead of arbitrary wait to ensure that it has loaded? Perhaps the table w result?

In our various testing env this 2000 might not be sufficient, so its safer to wait for some element to be loaded